src/Kiota.Builder/Writers/CSharp/CodeMethodWriter.cs

```diff
@@ -553,7 +551,15 @@ private static string GetBaseSuffix(bool isConstructor, bool inherits, CodeClass
                     thirdParameterName = $", {rawUrlParameter.Name}";
                 else if (parentClass.Properties.OfKind(CodePropertyKind.PathParameters) is CodeProperty pathParametersProperty && !string.IsNullOrEmpty(pathParametersProperty.DefaultValue))
                     thirdParameterName = $", {pathParametersProperty.DefaultValue}";
-                return $" : base({requestAdapterParameter.Name.ToFirstCharacterLowerCase()}, {urlTemplateProperty.DefaultValue}{thirdParameterName})";
+                if (currentMethod.Parameters.OfKind(CodeParameterKind.RequestAdapter) is CodeParameter requestAdapterParameter)
+                {
+                    return $" : base({requestAdapterParameter.Name.ToFirstCharacterLowerCase()}, {urlTemplateProperty.DefaultValue}{thirdParameterName})";
+                }
+                else if (parentClass.StartBlock?.Inherits?.Name?.Contains("Cli") == true)
```

Comment:
```suggestion
                else if (parentClass.StartBlock?.Inherits?.Name?.Contains("Cli", StringComparison.OrdinalIgnoreCase) == true)
```

Reply:
Class names aren't case insensitive. What's the rationale for this?

Reply:
In that case:

```suggestion
                else if (parentClass.StartBlock?.Inherits?.Name?.Contains("Cli", StringComparison.Ordinal) == true)
```

Reply:
A request builder that inherits client shouldn't be generated.

Reply:
Thanks for improving the check.

It doesn't have to be "Client" as a whole word because it is a substring search, "Cli" could be a substring to a lot of words. "Cli" is a substring for the following request builder type for Graph for example:

https://github.com/microsoftgraph/msgraph-cli/blob/main/src/generated/Identity/ApiConnectors/Item/UploadClientCertificate/UploadClientCertificateRequestBuilder.cs